src/mixins/object_stacking.mixin.js

```diff
@@ -76,5 +76,48 @@ fabric.util.object.extend(fabric.Object.prototype, /** @lends fabric.Object.prot
       this.canvas.moveTo(this, index);
     }
     return this;
+  },
+
+  /**
+   *
+   * @param {fabric.Object} other object to compare against
+   * @returns {boolean | undefined} if objects do not share a common ancestor or they are strictly equal it is impossible to determine which is in front of the other; in such cases the function returns `undefined`
+   */
+  isInFrontOf: function (other) {
+    if (this === other) {
+      return undefined;
```

Comment:
warn and return false?

Reply:
I think that it should remain.
The result of checking which object is in front of the other is really undefined. false means it is not.

Reply:
> I think that it should remain. The result of checking which object is in front of the other is really undefined in this case. false means it is not.

It may also mean that the ancestors to belong to any canvas yet.

Reply:
My reason to think is not true, is that isInFrontOf is added to support preserveObjectStacking: false with nested selections. is not a generic utility supposed to do help and behave correctly.
We need to support a use case, and we get to the length to say, what if developers uses it for other context and need to know if is true/false/undefined.

What if is private and no one can use it? simpler?

Reply:
I think it should be exposed publicly. That's why I made it work for all cases.
It's useful and will come to play in many place IMO

Reply:
perf-wise it's supposed to be great

Reply:
Yes it can be useful, but if you are using it wrongly i m not helping you.
We are covering:
- Object not on any canvas
-  me checking if myself is in front of me
that are in my opinion things that we can warn the developer about but not writing logic for.

We don't need to solve it now, but i ll get back to it.

Reply:
I didn't understand the last comment.
Take a closer look at the tests I wrote that cover this method.
I think that will explain best.